src/client/Client.cc

```diff
@@ -6649,11 +6649,17 @@ void Client::_unmount(bool abort)
   }
 
   mount_cond.wait(lock, [this] {
-    if (!mds_requests.empty()) {
-      ldout(cct, 10) << "waiting on " << mds_requests.size() << " requests"
-		     << dendl;
+    // Only wait for write OPs
+    for (auto& [tid, req] : mds_requests) {
+      if (req->is_write()) {
+        ldout(cct, 10) << "waiting for write request '" << tid
+                       << "' to complete, currently there are still "
+                       << mds_requests.size() << " read/write requests"
+                       << dendl;
```

Comment:
the line after comma looks a bit irrelevant, if that line is important piece of data then might be good to add it as 
`ldout(cct, 10) << "Current outstanding requests: " << mds_requests.size;()` before the `for` loop starts. something like
```
if(!mds_requests.empty()) {
    ldout(cct, 10) << "Current outstanding requests: " << mds_requests.size;   
}
```
and leave the line to just `ldout(cct, 10) << tid << " waiting for write request to complete";`

what do you think?

Reply:
> the line after comma looks a bit irrelevant, if that line is important piece of data then might be good to add it as `ldout(cct, 10) << "Current outstanding requests: " << mds_requests.size;()` before the `for` loop starts. something like
> 
> ```
> if(!mds_requests.empty()) {
>     ldout(cct, 10) << "Current outstanding requests: " << mds_requests.size;   
> }
> ```

This will print a log if the `mds_requests` is non-empty but they are all `read` requests, which seems unnecessary here.

> 
> and leave the line to just `ldout(cct, 10) << tid << " waiting for write request to complete";`
> 

This seems confusing for me. `waiting for write request '${tid}' to complete` looks much better.

> what do you think?

Reply:
> This will print a log if the mds_requests is non-empty but they are all read requests, which seems unnecessary here.

makes sense

> This seems confusing for me. waiting for write request '${tid}' to complete looks much better.

Yeah this is better